app/controllers/auth/google_controller.rb

```diff
@@ -72,7 +73,9 @@ def register_with_google(name, email, role, access_token, google_id)
     else
       @user = User.find_or_initialize_by(email: email)
       if @user.new_record?
-        new_google_user(name, email, role, access_token, google_id, @user)
+        if !new_google_user(name, email, role, access_token, google_id, @user)
```

Comment:
I'm not sure if I like this (there's probably a better way to do this), but this would allow `register_with_google` to manage all of the redirects and avoid a double-redirect in the case where `new_google_user` doesn't actually save the user.

I want to test this on staging first to make sure it works properly though.

Reply:
Okay, looks like this works alright.